DailyReportWeb/app/report.js

```diff
@@ -6,17 +6,46 @@
             function reportCtrl($scope, $http, $routeParams) {
                 var ctrl = this;
                 ctrl.instanceId = $routeParams.instanceId;
-                ctrl.isLoading = true;
                 ctrl.data = {};
                 ctrl.actions = {};
 
-                $http.get("/api/report/" + ctrl.instanceId)
-                    .success(function (data) {
-                        ctrl.data = data;
-                    })
-                    .finally(function () {
-                        ctrl.isLoading = false;
-                    });
+                ctrl.actions.getDashboardData = function () {
+                    ctrl.isLoading = true;
+
+                    $http.get("/api/report/" + ctrl.instanceId)
+                        .success(function (data) {
+                            ctrl.data = formatDates(data);
+                        })
+                        .finally(function () {
+                            ctrl.isLoading = false;
+                        });
+
+                    function formatDates(data) {
+                        var i, j, k;
+
+                        for (i = 0; i < data.length; i++) {
+                            for (j = 0; j < data[i].worklogs.length; j++) {
+                                data[i].worklogs[j].dayHumanReadable = moment(data[i].worklogs[j].date).format("DD/MMM");
```

Comment:
This should be converted directly in the view.  Do not iterate through all of them and set them.

Reply:
Are you saying that I should have conversation methods and use them in the view, instead of iterating through each element in the UI? Is your way faster or what is the reason behind moving the logic to the view?

Reply:
HOW you display a date is the presentation's layer task. In AngularJS there are 'filters' that help you achieve this part.
No other part of the app needs to know the human readable representation of the date. the rest of the components work with Date types.

also, look at the following code in the UI, rather than having all those for loops in the javascript file:
```
1. <span>{{ dashboardItem.worklogs[0].date | neatDate }}<span>
2. <span>{{ dashboardItem.worklogs[0].totalTimeSpentInSeconds | neatDuration }}<span>
```

Then, if for example you want to change how the duration is displayed, you'll just change the implementation of the respective filter, and automatically all times are correctly displayed.